mypyc/ir/ops.py

```diff
@@ -799,6 +799,24 @@ def accept(self, visitor: OpVisitor[T]) -> T:
         return visitor.visit_get_attr(self)
 
 
+class GetAttrNullable(GetAttr):
```

Comment:
It would be more consistent with other ops to add a new flag to `GetAttr` instead of adding a subclass. For example, add `allow_null: bool = False` as a keyword-only argument to `GetAttr.__init__`, and set the error kind accordingly in `__init__` when the flag is true, and a boolean attribute to `GetAttr`. I expect that this change would also reduce the the size of the PR. Can you make this change?

(Your approach clearly has some benefits over adding a flag, as it's arguably easy to forget to handle the flag in a visitor, but since we already have a bunch of flags that impact the semantics in `Op` subclasses, this is something that already needs to be remembered when writing a visitor.)

Reply:
I expect the performance impact to be quite small, since checking a boolean flag is quick (here branch prediction success rate should be high). I try to avoid subclasses of concrete classes such as `GetAttr` as a style convention, especially in new code, so I'd prefer to have this as a flag.